lib/utils/proxy/proxy.go

```diff
@@ -64,20 +67,45 @@ func (d directDial) dialALPNWithDeadline(network string, addr string, config *ss
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	tlsConn, err := tls.DialWithDialer(dialer, network, addr, conf)
+
+	tlsDialer := tls.Dialer{
+		NetDialer: dialer,
+		Config:    conf,
+	}
+
+	tlsConn, err := tlsDialer.DialContext(ctx, network, addr)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	return sshutils.NewClientConnWithDeadline(tlsConn, addr, config)
+	return NewClientConnWithDeadline(ctx, tlsConn, addr, config)
+}
+
+// NewClientConnWithDeadline establishes new client connection with specified deadline
+func NewClientConnWithDeadline(ctx context.Context, conn net.Conn, addr string, config *ssh.ClientConfig) (*ssh.Client, error) {
```

Comment:
it looks like `(api/utils/sshutils).NewClientConnWithDeadline` is untouched in this PR, you're just replacing it at some of the call sites with this version with calls the new `(lib/srv/sshutils).NewClientConn`. Is there a reason not to modify the original `api` version? Should all teleport code call the new traced version, or are there cases to use the other one?